managed/services/backup/pitr_timerange_service.go

```
@@ -0,0 +1,357 @@
+	Start      uint32 `json:"start"`
+	End        uint32 `json:"end"`
+	Size       int64  `json:"-"`
+}
+
+type gap struct {
+	s, e uint32
+}
+
+type gaps []gap
+
+func (x gaps) Len() int { return len(x) }
+func (x gaps) Less(i, j int) bool {
+	return x[i].s < x[j].s || (x[i].s == x[j].s && x[i].e < x[j].e)
+}
+func (x gaps) Swap(i, j int) { x[i], x[j] = x[j], x[i] }
+
+// compressionType is the type of compression used for PITR oplog
+type compressionType string
+
+const (
+	compressionTypeNone      compressionType = "none"
+	compressionTypePGZIP     compressionType = "pgzip"
+	compressionTypeSNAPPY    compressionType = "snappy"
+	compressionTypeLZ4       compressionType = "lz4"
+	compressionTypeS2        compressionType = "s2"
+	compressionTypeZstandard compressionType = "zstd"
+)
+
+// file return compression alg based on given file extension
+func file(ext string) compressionType {
+	switch ext {
+	case "gz":
+		return compressionTypePGZIP
+	case "lz4":
+		return compressionTypeLZ4
+	case "snappy":
+		return compressionTypeSNAPPY
+	case "s2":
+		return compressionTypeS2
+	case "zst":
+		return compressionTypeZstandard
+	default:
+		return compressionTypeNone
+	}
+}
+
+func (ss *PITRTimerangeService) getPITROplogs(ctx context.Context, location *models.BackupLocation, artifactName string) ([]*oplogChunk, error) {
+	if location.S3Config == nil {
+		return nil, errUnsupportedLocation
+	}
+
+	var err error
+	var oplogChunks []*oplogChunk
+
+	prefix := path.Join(artifactName, pitrFSPrefix)
+	pitrFiles, err := ss.locationClient.List(ctx, location.S3Config.Endpoint, location.S3Config.AccessKey, location.S3Config.SecretKey, location.S3Config.BucketName, prefix, "")
+	if err != nil {
+		return nil, errors.Wrap(err, "get list of pitr chunks")
+	}
+	if len(pitrFiles) == 0 {
+		return nil, nil
+	}
+
+	for _, f := range pitrFiles {
+		if f.IsDeleteMarker {
+			ss.l.Warningf("skip pitr chunk %s/%s because of file has delete marker", prefix, f.Name)
```

Why it's warning? How user should react to that? I guess it's normal situation when some files marked for deletion and we can just silently ignore them.

---

We can set `debug` severity to this message.